magellan-library/src/main/java/com/wealthfront/magellan/view/DialogComponent.kt

```diff
@@ -0,0 +1,56 @@
+package com.wealthfront.magellan.view
+
+import android.app.Dialog
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import com.wealthfront.magellan.lifecycle.LifecycleAware
+import javax.inject.Inject
+
+class DialogComponent @Inject constructor() : LifecycleAware {
+
+  private var dialogCreator: DialogCreator? = null
+  private var dialog: Dialog? = null
+  private var context: Context? = null
+
+  @VisibleForTesting internal var dialogIsShowing: Boolean = false
+
+  fun showDialog(dialogCreator: DialogCreator) {
+    this.dialogCreator = dialogCreator
+    this.dialogIsShowing = true
+    createDialog()
+  }
+
+  override fun create(context: Context) {
+    this.context = context
+  }
+
+  override fun show(context: Context) {
+    createDialog()
+  }
+
+  override fun hide(context: Context) {
+    destroyDialog()
+  }
+
+  override fun destroy(context: Context) {
+    this.context = null
+  }
+
+  private fun createDialog() {
+    if (dialogCreator != null && context != null && dialogIsShowing) {
+      dialog = dialogCreator!!.invoke(context!!)
+      dialog!!.show()
+    }
+  }
+
+  private fun destroyDialog() {
+    if (dialog != null) {
+      dialogIsShowing = dialog!!.isShowing
+      dialog!!.setOnDismissListener(null)
+      dialog!!.dismiss()
+      dialog = null
+    }
+  }
+}
+
+typealias DialogCreator = (Context) -> Dialog
```

Comment:
nit: might want to put this at the top of the file, feels kind of weird dangling at the bottom haha.

Reply:
d